app/models/concerns/foreman_remote_execution/errors_flattener.rb

```diff
@@ -0,0 +1,62 @@
+module ForemanRemoteExecution
+  module ErrorsFlattener
+    def flattened_validation_exception
+      ActiveRecord::RecordNotSaved.new(I18n.t("activerecord.errors.messages.record_invalid", :errors => flattened_errors.join(', ')))
+    end
+
+    def flattened_errors
+      errors = Hash.new { |h, k| h[k] = [] }
+      self.errors.keys.each do |key|
+        messages = self.errors[key]
+        invalid_objects = invalid_objects_for_attribute(key)
+        if invalid_objects.blank?
+          errors[key] = messages
+        else
+          invalid_objects.each do |invalid_object|
+            errors.merge!(sub_object_errors(key, invalid_object, messages))
+          end
+        end
+      end
+      errors.map { |key, messages| self.errors.full_message(key, messages.join(', ')) }
+    end
+
+    private
+
+    def invalid_objects_for_attribute(attribute)
+      if self.respond_to?(attribute)
+        invalid_object = self.send(attribute)
```

Comment:
Good point, new rubocop would have found that as well https://github.com/bbatsov/rubocop/commit/242a761e0cafc2e4634bd62a3362bed2b70fcec2 (but we need to rely on the one in foreman right now)